p4-fusion/commands/file_data.cc

```diff
@@ -53,24 +59,43 @@ void FileData::SetFromDepotFile(const std::string& fromDepotFile, const std::str
 	}
 }
 
-void FileData::MoveContentsOnceFrom(const std::vector<char>& contents)
+void FileData::StartWrite()
 {
-	// TODO double-check the thread logic here.  It needs to be thread safe.
+	MTR_SCOPE("FileData", __func__);
 
-	if (m_data->isContentsSet)
+	if (m_data->isBlobOIDSet)
 	{
 		// Do not set the contents.  Assume that
 		// they were already set or, worst case, are currently being set.
 		return;
 	}
-	m_data->isContentsSet = true;
-	m_data->contents = std::move(contents);
+
+	GIT2(git_repository_open_bare(&repo, FileData::repoPath.c_str()));
```

Comment:
TODO: We shouldn't need to access repoPath from a global here, and potentially also shouldn't need to open the repository again every time, maybe we can just pass the repo handle down to here.

Reply:
fixed